cli/bench/bench.go

```diff
@@ -0,0 +1,140 @@
+package bench
+
+import (
+	bctx "context"
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/FZambia/tarantool"
+	"github.com/tarantool/cartridge-cli/cli/context"
+)
+
+func printResults(results Results) {
+	fmt.Printf("\nResults:\n")
+	fmt.Printf("\tSuccess operations: %d\n", results.successResultCount)
+	fmt.Printf("\tFailed  operations: %d\n", results.failedResultCount)
+	fmt.Printf("\tRequest count: %d\n", results.handledRequestsCount)
+	fmt.Printf("\tTime (seconds): %f\n", results.duration)
+	fmt.Printf("\tRequests per second: %d\n\n", results.requestsPerSecond)
+}
+
+func spacePreset(ctx context.BenchCtx, tarantoolConnection *tarantool.Connection) error {
+	dropSpace(ctx, tarantoolConnection)
+	return createSpace(tarantoolConnection)
+}
+
+func incrementRequest(err error, results *Results) {
+	if err == nil {
+		results.successResultCount++
+	} else {
+		results.failedResultCount++
+	}
+	results.handledRequestsCount++
+}
+
+func connectionLoop(ctx context.BenchCtx, tarantoolConnection *tarantool.Connection, requestsLoop requestLoopType, results *Results, backgroundCtx bctx.Context) {
+	var connectionWait sync.WaitGroup
+	for i := 0; i < ctx.SimultaneousRequests; i++ {
+		connectionWait.Add(1)
+		go func() {
+			defer connectionWait.Done()
+			requestsLoop(ctx, tarantoolConnection, results, backgroundCtx)
+		}()
+	}
+
+	connectionWait.Wait()
+}
+
+func Run(ctx context.BenchCtx) error {
+	rand.Seed(time.Now().UnixNano())
+
+	operationsMap := []Operation{
+		{"insert", ctx.InsertCount, insertRequestsLoop, 0, 0},
+		{"delete", ctx.DeleteCount, deleteRequestsLoop, 0, 0},
+		{"update", ctx.UpdateCount, updateRequestsLoop, 0, 0},
+		{"select", ctx.SelectCount, selectRequestsLoop, 0, 0},
+	}
+
+	allOperationCount := 0
+	for _, operation := range operationsMap {
+		allOperationCount += operation.count
+	}
+
+	connectionsSum := 0
+	maxConnectionsCount := ctx.Connections
+	for i, _ := range operationsMap {
+		operationsMap[i].loadProfilePercent = float64(operationsMap[i].count) / float64(allOperationCount)
+		operationsMap[i].connectionsCount = int(float64(maxConnectionsCount) * operationsMap[i].loadProfilePercent)
+		if operationsMap[i].count != 0 && operationsMap[i].connectionsCount == 0 {
+			operationsMap[i].connectionsCount = 1
+		}
+		connectionsSum += operationsMap[i].connectionsCount
+	}
+
+	if connectionsSum > maxConnectionsCount {
+		return fmt.Errorf("Use %d connections count for this load profile\n", connectionsSum)
```

Comment:
Set a dot at the end of the sentence.
As for me, I preferred more verbose errors. Really, it's not so clear what is wrong.
```
┌─[leonid@vasya-hp]─[~/work/mail/cartridge-cli]
└──╼ ./cartridge bench --insert 1 --select 0 --update 1 --delete 0 --connections 1
   ⨯ Use 2 connections count for this load profile
```
Up to you.

Reply:
This feature dropped right now